clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -2132,7 +2133,7 @@ mlir::Value CIRGenFunction::buildAlloca(StringRef name, mlir::Type ty,
                                         mlir::Location loc, CharUnits alignment,
                                         bool insertIntoFnEntryBlock) {
   mlir::Block *entryBlock = insertIntoFnEntryBlock
-                                ? &CurFn.getRegion().front()
+                                ? &CurFn->getRegion(0).front()
```

Comment:
Can you abstract this into a CIRGenModule method `getCurFunctionEntryBlock` and assert if CurFn isn't a FuncOp. In case those global ctors end up having allocas, we'll hit the assert and implement the proper region to get at that point - I'm afraid hardcoding `getRegion(0)` here could be setting us a trap in the future.